photosynthesis_metrics/brisque.py

```diff
@@ -32,16 +36,25 @@ def _ggd_parameters(x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
 
 def _aggd_parameters(x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
     gamma = torch.arange(start=0.2, end=10.001, step=0.001).to(x)
-    r_table = torch.exp(2 * torch.lgamma(2. / gamma) - torch.lgamma(1. / gamma) - torch.lgamma(3. / gamma)).repeat(
-        x.size(0), 1)
+    r_table = torch.exp(2 * torch.lgamma(2. / gamma) - torch.lgamma(1. / gamma) - torch.lgamma(3. / gamma))
+    r_table = r_table.repeat(x.size(0), 1)
 
     mask_left = x < 0
     mask_right = x > 0
-    count_left = mask_left.sum(dim=(-1, -2))
-    count_right = mask_right.sum(dim=(-1, -2))
+    count_left = mask_left.sum(dim=(-1, -2), dtype=torch.float32)
+    count_right = mask_right.sum(dim=(-1, -2), dtype=torch.float32)
+
+    assert (count_left > 0).all(), 'Expected input tensor (pairwise products of neighboring MSCN coefficients)' \
```

Comment:
This is not what a typical user expects from a metric. Currently, I should check all images manually before passing to a metric. Since I'm gonna use it during GANs training, sometimes images can not meet this criterion. Why not add EPS? In this case, I won't have to worry about it.

Reply:
I agree that this is not a typical experience. In the first commit I added EPS to avoid nan. However, after closer look at BRISQUE theory, I decided to add asserts for now. This results from the estimation of parameters of generalised gaussian distribution where \sigma should be strictly greater zero. In case of images with \sigma = 0, we deal with Dirac function instead of GGD. Same applies to AGGD. 

In addition available implementations return NaN as well. 
For now, I would suggest to assert such inputs in accordance to theory instead of returning undefined finite number.

Reply:
@PavelOstyakov I agree with @denproc that undefined finite number in case of a zero-filled tensor input does not make much sense. 
I would also argue that zero-filled tensor is quite a unique thing in a real setting in general and in GAN training in particular. May be we do not see something and you could provide a bit more context on why the behavior that you propose is strictly better than the one which @denproc proposed?